File: merge-overlapping-intervals/MergeOverlappingInterval.java
import java.util.*;

public class MergeOverlappingInterval {
    public int[][] mergeOverlappingIntervals2(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> a[0] - b[0]);
        final int START_TIME = 0;
        final int END_TIME = 1;
        List<int[]> result = new ArrayList<>(); result.add(intervals[0]);
        for(int i = 1; i < intervals.length; i++) {
            int lastIntervalIndex = result.size() - 1;
            int[] current = intervals[i];
            int[] previous = result.get(lastIntervalIndex);

            boolean isOverlap = previous[END_TIME] >= current[START_TIME];
            if(isOverlap) {
                previous[END_TIME] = Math.max(current[END_TIME], previous[END_TIME]);
                result.set(lastIntervalIndex, previous);
                continue;
            }
            result.add(current);
        }
        return result.toArray(new int[result.size()][]);
    }
}

File: merge-overlapping-intervals/MergeOverlappingInterval.swift
import Foundation

class MergeOverlappingInterval {
    let START_TIME: Int = 0
    let END_TIME: Int = 1
    func mergeOverlappingIntervals(_ intervals: [[Int]]) -> [[Int]] {
        var interval = intervals.sorted(by: { $0[0] < $1[0] })
        var overlaps: [[Int]] = []
        overlaps.append(interval.first!)
        for index in stride(from: 1, to: interval.count, by: 1) {
        var previous = overlaps.last!
        var current = interval[index]
        let isOverlap = previous[END_TIME] >= current[START_TIME]
        if(isOverlap) {
            overlaps[overlaps.count - 1][1] = max(previous[END_TIME], current[END_TIME])
            continue
        }
        overlaps.append(current)
        }
        return overlaps;
    }
}